primitiv/c/graph.cc

```diff
@@ -7,224 +7,193 @@
 #include <vector>
 
 #include <primitiv/graph.h>
-
 #include <primitiv/c/internal.h>
 #include <primitiv/c/graph.h>
 
 using primitiv::Node;
 using primitiv::Graph;
+using primitiv::c::internal::to_c_ptr;
+using primitiv::c::internal::to_cpp_ptr;
+using primitiv::c::internal::to_c_ptr_from_value;
 
 extern "C" {
 
 primitiv_Node *primitiv_Node_new() {
-  return to_c(new Node);
-}
-primitiv_Node *safe_primitiv_Node_new(primitiv_Status *status) {
-  SAFE_RETURN(primitiv_Node_new(), status, nullptr);
+  return to_c_ptr(new Node);
 }
 
-primitiv_Node *primitiv_Node_new_with_movement(primitiv_Node *node) {
-  return to_c(new Node(std::move(*to_cc(node))));
-}
-primitiv_Node *safe_primitiv_Node_new_with_movement(primitiv_Node *node,
-                                                    primitiv_Status *status) {
-  SAFE_RETURN(primitiv_Node_new_with_movement(node), status, nullptr);
+primitiv_Status primitiv_Node_new_from_node(primitiv_Node *src,
+                                            primitiv_Node **node) {
+  try {
+    *node = to_c_ptr(new Node(std::move(*to_cpp_ptr(src))));
+    return ::primitiv_Status::PRIMITIV_OK;
+  } HANDLE_EXCEPTION
 }
 
 void primitiv_Node_delete(primitiv_Node *node) {
-  delete to_cc(node);
-}
-void safe_primitiv_Node_delete(primitiv_Node *node, primitiv_Status *status) {
-  SAFE_EXPR(primitiv_Node_delete(node), status);
+  delete to_cpp_ptr(node);
 }
 
 bool primitiv_Node_valid(const primitiv_Node *node) {
```

Comment:
Use `_Bool` instead of `bool`. In C, `bool` is an alias of `_Bool` defined in `<stdbool.h>`.